test/test_device_select_unique_by_key.cu

```diff
@@ -0,0 +1,741 @@
+        typeid(KeyT).name(),
+        (int) sizeof(ValueT));
+    fflush(stdout);
+
+    // Run Test
+    Test<BACKEND>(h_keys_in, h_values_in, h_keys_reference, h_values_reference, num_selected, num_items);
+
+    // Cleanup
+    if (h_keys_reference) delete[] h_keys_reference;
+    if (h_values_reference) delete[] h_values_reference;
+}
+
+
+/**
+ * Test different gen modes
+ */
+template <
+    Backend         BACKEND,
+    typename        KeyT,
+    typename        ValueT>
+void Test(
+    int             num_items)
+{
+    for (int max_segment = 1; ((max_segment > 0) && (max_segment < num_items)); max_segment *= 11)
+    {
+        TestPointer<BACKEND, KeyT, ValueT>(num_items, 0, max_segment);
+        TestPointer<BACKEND, KeyT, ValueT>(num_items, 2, max_segment);
+        TestPointer<BACKEND, KeyT, ValueT>(num_items, 7, max_segment);
+    }
+}
+
+
+/**
+ * Test different dispatch
+ */
+template <
+    typename        KeyT,
+    typename        ValueT>
+void TestOp(
+    int             num_items)
+{
+    Test<CUB, KeyT, ValueT>(num_items);
+#ifdef CUB_CDP
+    Test<CDP, KeyT, ValueT>(num_items);
+#endif
+}
+
+
+/**
+ * Test different input sizes
+ */
+template <
+    typename        KeyT,
+    typename        ValueT>
+void Test(
+    int             num_items)
+{
+    if (num_items < 0)
+    {
+        TestOp<KeyT, ValueT>(0);
+        TestOp<KeyT, ValueT>(1);
+        TestOp<KeyT, ValueT>(100);
+        TestOp<KeyT, ValueT>(10000);
+        TestOp<KeyT, ValueT>(1000000);
+    }
+    else
+    {
+        TestOp<KeyT, ValueT>(num_items);
+    }
+}
+
+
+
+//---------------------------------------------------------------------
+// Main
+//---------------------------------------------------------------------
+
+/**
+ * Main
+ */
+int main(int argc, char** argv)
+{
+    int num_items           = -1;
+    int entropy_reduction   = 0;
+    int maxseg              = 1000;
+
+    // Initialize command line
+    CommandLineArgs args(argc, argv);
+    g_verbose = args.CheckCmdLineFlag("v");
+    args.GetCmdLineArgument("n", num_items);
+    args.GetCmdLineArgument("i", g_timing_iterations);
+    args.GetCmdLineArgument("repeat", g_repeat);
+    args.GetCmdLineArgument("maxseg", maxseg);
+    args.GetCmdLineArgument("entropy", entropy_reduction);
+
+    // Print usage
+    if (args.CheckCmdLineFlag("help"))
+    {
+        printf("%s "
+            "[--n=<input items> "
+            "[--i=<timing iterations> "
+            "[--device=<device-id>] "
+            "[--maxseg=<max segment length>]"
+            "[--entropy=<segment length bit entropy reduction rounds>]"
+            "[--repeat=<repetitions of entire test suite>]"
+            "[--v] "
+            "[--cdp]"
+            "\n", argv[0]);
+        exit(0);
+    }
+
+    // Initialize device
+    CubDebugExit(args.DeviceInit());
+    g_device_giga_bandwidth = args.device_giga_bandwidth;
+    printf("\n");
+
+#ifdef CUB_TEST_MINIMAL
```

Comment:
These are no longer used as of #403. See [these changes](https://github.com/NVIDIA/cub/pull/403/files#diff-95a45d0e7312e0f0952a14a791cafac27ef85578189fa38492d1a6d2adabcdf7) to `test_device_select_unique.cu` that are aimed at reducing build/run times. This should be updated to use the same patterns.